lib/kubeclient/watch_stream.rb

```diff
@@ -1,54 +1,63 @@
 require 'json'
-require 'net/http'
+require 'http'
 module Kubeclient
   module Common
     # HTTP Stream used to watch changes on entities
     class WatchStream
       def initialize(uri, http_options, format: :json)
         @uri = uri
-        @http = nil
-        @http_options = http_options.merge(read_timeout: nil)
+        @http_client = nil
+        @http_options = http_options
         @format = format
       end
 
       def each
         @finished = false
-        @http = Net::HTTP.start(@uri.host, @uri.port, @http_options)
 
-        buffer = ''
-        request = generate_request
+        @http_client = build_client
+        response = @http_client.request(:get, @uri, build_client_options)
+        unless response.code < 300
```

Comment:
@simon3z, unfortunately `HTTP` does not have a fancy `Success` wrapper class and deals only with raw integer codes. In case of `net/http` the trick is to have all the 2xx codes inherit from `Net::HTTPSuccess`, see [here](https://github.com/ruby/ruby/blob/trunk/lib/net/http/responses.rb#L36). So in our case the only difference is that we treat 1xx responses as success as well. I can fix this if you think it makes a difference.